scripts/globals/weaponskills.lua

```diff
@@ -250,7 +251,31 @@ local function getMultiAttacks(attacker, target, wsParams)
         end
     end
 
-    numHits = utils.clamp(numHits + bonusHits, 1, 8)
+    -- If this WS/Ability is capable of multi-striking from Occasionally Attacks X weapons, calculate now
+    -- Note: QA/TA/DA take precedence over OAX, so if they proc we skip.
+    local totalOAXHits = 0
+    if
+        canOAX and
+        bonusHits <= 0
+    then
+        local numMainHits = attacker:getWeaponHitCount()
+        -- Subtract 1 since the initial hit would have been calculated already; we just want additionals
+        numMainHits = utils.clamp(numMainHits - 1, 0, numMainHits)
```

Comment:
Do we need this clamp if numMainHits will always be at least 1 or can it be 0? Same for the offhand clamp

Reply:
Just verified; the value ultimately comes from battleutils.cpp "getHitCount" which is indeed minimum of 1 and already caps at 8 in the return.

I'll remove the redundant clamps.